### DISCUSSION TITLE
Support for Nerd Fonts v3+

### DISCUSSION THREAD

**Discussion by acidghost** [jarun/nnn#1635] [Ideas]
Would be nice to support newer releases of Nerd Fonts.

If you want icons with nnn you'll have to stick to an old release of Nerd Fonts.

**Comment by jarun** [comment]
@luukvbaal @N-R-K @KlzXS @0xACE how about dropping support for Nerd Fonts? I find Emoji support stabler and consistent across distros.

Please share your opinion.

**Reply by luukvbaal** [reply]
I haven't tried the emoji option TBH. I've been satisfied with/liked the look of the nerd-font icons. 

It's also the more popular version on the AUR: https://aur.archlinux.org/packages?O=0&K=nnn. I think nerd-fonts are quite popular in dev environments in general. Dropping support for it would be a shame I think, but it's unfortunate that they decided to break backwards compatibility in their new release.

**Reply by jarun** [reply]
Yes, you are right it's the most popular one out there. Let me think about it while others respond.

**Reply by jarun** [reply]
BTW, I like emojis because they work out of the box for me on Ubuntu and Termux. I didn't have to install any additional package and because of the widespread use of Emojis the codes are pretty much standard.

**Reply by luukvbaal** [reply]
You still need a font that supports it but yeah it will be more common the be included by default on distros than nerd-font. 

Having just tried it I must admit the emojis look quite good as well in the file manager. But there are also a several popular (n)vim plugins I use that all use nerd-font icons and I like the consistency of having them in nnn as well.

**Reply by WanderLanz** [reply]
Given how ubiquitous they are, I'm more inclined to suggest user preference and just continue supporting all (Terminal, Emojis, Nerd Fonts, Nerd Fonts V3), but I totally understand not wanting to have to deal with incompatible nerd font versions.

Just my 2 dumb cents.

I'll link the [translation table](https://github.com/ryanoasis/nerd-fonts/issues/1059#issuecomment-1404891287) here just in case anyone loses it.

**Comment by N-R-K** [comment]
@jarun I like the nerd icons (which is what I use on my personal nnn build) more than the emojies because they're more detailed and look nicer. As such, I would ask not to drop nerd font support.

As for nerd font v3... it's a shame that they decided to break backwards compatibility like this.

Do we have any numbers on just how many icons are broken @luukvbaal ? If it's not too many then supporting both v2 and v3 should be feasible, probably.

**Reply by luukvbaal** [reply]
I don't, I just noticed that like half of the icons in my home folder are missing after having updated my font.

**Comment by luukvbaal** [comment]
This might prove useful: https://github.com/loichyan/nerdfix

**Comment by jarun** [comment]
Can someone confirm if they have a history of doing this?

**Reply by dorsiflexion** [reply]
It's [seems to be the first](https://github.com/ryanoasis/nerd-fonts/releases?q=break&expanded=true) breaking change that broke something noteworthy.
The last and only other breaking change was 2018.

**Reply by jarun** [reply]
@N-R-K @luukvbaal in that case can we support only Nerd Fonts v3+ from the next release and move on?
Can v3+ be installed anywhere? Are there any library deps?

**Reply by N-R-K** [reply]
> Can v3+ be installed anywhere? Are there any library deps?

There are no library deps. You just need to install the font, and then select it as your terminal font - that's all.

> can we support only Nerd Fonts v3+ from the next release and move on?

If the number of breaking icons is large then dropping support for v2 and only supporting v3 is fine by me.

**Reply by luukvbaal** [reply]
Fine by me. Nope it's just a font. It can be updated through package manager on some distros but even downloading the font file from their download page and installing it is fairly standard practice. Shouldn't be much of a problem if we just support v3 I think.

**Reply by jarun** [reply]
> It can be updated through package manager on some distros but even downloading the font file from their download page and installing it is fairly standard practice.

Yes, the manual installation should be easy enough since package managers on several distros take long to update to latest releases for stability.

**Reply by jarun** [reply]
One of you please work on it.

**Reply by luukvbaal** [reply]
Working on it.

**Reply by jarun** [reply]
Thank you!

**Comment by jarun** [comment]
Closing since this is in now.
